DatadogSessionReplay/Sources/Writers/ResourcesWriter.swift

```diff
@@ -16,25 +16,99 @@ internal protocol ResourcesWriting {
 
 internal class ResourcesWriter: ResourcesWriting {
     /// An instance of SDK core the SR feature is registered to.
-    private weak var core: DatadogCoreProtocol?
+    private let scope: FeatureScope?
+    private let telemetry: Telemetry
+
+    @ReadWriteLock
+    private var knownIdentifiers = Set<String>() {
+        didSet {
+            if let knownIdentifiers = knownIdentifiers.asData() {
+                scope?.dataStore.setValue(
+                    knownIdentifiers,
+                    forKey: Constants.knownResourcesKey
+                )
+            }
+        }
+    }
 
     init(
-        core: DatadogCoreProtocol
+        core: DatadogCoreProtocol,
+        dataStoreResetTime: TimeInterval = TimeInterval(30).days
     ) {
-        self.core = core
+        self.scope = core.scope(for: ResourcesFeature.self)
+        self.telemetry = core.telemetry
+
+        self.scope?.dataStore.value(forKey: Constants.storeCreationKey) { result in
+            if let storeCreation = result.data()?.asTimeInterval(), Date().timeIntervalSince1970 - storeCreation < dataStoreResetTime {
+                self.scope?.dataStore.value(forKey: Constants.knownResourcesKey) { [weak self] result in
+                    switch result {
+                    case .value(let data, _):
+                        if let knownIdentifiers = data.asKnownIdentifiers() {
+                            self?.knownIdentifiers.formUnion(knownIdentifiers)
+                        }
+                    case .error(let error):
+                        self?.telemetry.error("Failed to read processed resources from data store: \(error)")
+                    case .noValue:
+                        break
+                    }
+                }
+            } else { // Reset if store was created more than 30 days ago
+                self.scope?.dataStore.setValue(
+                    Date().timeIntervalSince1970.asData(),
+                    forKey: Constants.storeCreationKey
+                )
+                self.scope?.dataStore.removeValue(forKey: Constants.knownResourcesKey)
+            }
+        }
     }
 
     // MARK: - Writing
 
     func write(resources: [EnrichedResource]) {
-        guard let scope = core?.scope(for: ResourcesFeature.self) else {
-            return
-        }
-        scope.eventWriteContext { _, recordWriter in
-            resources.forEach {
-                recordWriter.write(value: $0)
+        scope?.eventWriteContext { [weak self] _, recordWriter in
+            let unknownResources = resources.filter { self?.knownIdentifiers.contains($0.identifier) == false }
+            for resource in unknownResources {
+                print("👾 Resource ID: ", resource.identifier)
+                recordWriter.write(value: resource)
             }
+            self?.knownIdentifiers.formUnion(Set(unknownResources.map { $0.identifier }))
+        }
+    }
+
+    enum Constants {
```

Comment:
this and below are `internal` to be able to access them in tests (avoids reimplementing or retyping)